### PR TITLE
feat: write/log report in suite

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

/area conformance

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes-sigs/gateway-api/issues/2289

**Does this PR introduce a user-facing change?**:

```release-note
support automatically dump report in test suite
```

### CODE REVIEW

**Review by sunjayBhatia — APPROVED** [review]
Looks like it works as advertised 👍🏽, was able to get a report YAML out using the golang suite

File: conformance/experimental_conformance_test.go [hunk 1]
```diff
@@ -119,32 +117,16 @@ func testExperimentalConformance(t *testing.T) {
...
-	}
-
-	if output != "" {
-		if err = os.WriteFile(output, rawReport, 0644); err != nil {
-			return err
-		}
-	}
-	logf("Conformance report:\n%s", string(rawReport))
-
-	return nil
```
Comment (youngnick) [inline conformance/experimental_conformance_test.go:149]: Why move all of this? You could also achieve this by passing `cSuite.ReportOutput` to the `writeReport` call on line 133, which would keep the output logic in the containing block, keeping the `Report` call more simple and closer to single-purpose.

File: conformance/utils/suite/suite.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ func New(s Options) *ConformanceTestSuite {
 	}
 
 	switch {
-	case s.EnableAllSupportedFeatures == true:
```
Comment (youngnick) [inline conformance/utils/suite/suite.go:92]: Seems likely to be an IDE edit? It's fine in this case, but in general it's better to put this sort of change in a separate PR.

**Review by youngnick — CHANGES_REQUESTED** [review]
I think that it's better to leave `writeReports` in place, and just change what we pass into it, rather than moving it inside the `Report` method.

File: conformance/utils/suite/suite.go (hunk 2 above)
Comment (Xunzhuo) [reply conformance/utils/suite/suite.go:93]: Yep: https://github.com/kubernetes-sigs/gateway-api/pull/2346

File: conformance/experimental_conformance_test.go (hunk 1 above)
Comment (Xunzhuo) [reply conformance/experimental_conformance_test.go:152]: My idea is that, when gwapi implementations want to get the Report and use the experiment test suite, the flag `cSuite.Report` is not used inside the `experiment test suite` itself, we need to rewrite another writeReport to achieve this.
